schema.go

```diff
@@ -48,6 +48,18 @@ type RootSchema struct {
 	SchemaURI string `json:"$schema"`
 }
 
+// TopLevelType returns a string representing the schema's top-level type.
+func (rs *RootSchema) TopLevelType() string {
+	validator, ok := rs.Schema.Validators["type"]
```

Comment:
This can be shortened to:
```go
if t, ok := rs.Schema.Validators["type"].(*Type); ok {
	return t.String()
}
```

Reply:
Ah, I didn't realize that the ok could be chained like that, handling both possible errors.